sydent/db/valsession.py

```diff
@@ -121,6 +121,64 @@ def getTokenSessionById(self, sid):
 
         return None
 
+    def next_link_differs(self, sid, token, next_link):
+        """Check whether the specified session has already been validated with a
+        next_link provided, and if  next_link value
+
+        :param sid: The session ID
+        :type sid: str
+
+        :param token: The validation token
+        :type token: str
+
+        :param next_link: The next_link parameter used in submitting this validation
+        :type next_link: str
+
+        :returns: Whether the provided next_link differs from the provided
+            token's associated next_link. Returns False if the stored next_link
+            value is NULL.
+        :rtype: bool
+        """
+        cur = self.sydent.db.cursor()
+
+        # Check if this session has already been validated
+        s = self.getTokenSessionById(sid)
+        if not s.validated:
+            # This session has not been validated before, we allow it to be
+            return False
+
+        cur.execute("select next_link_used from threepid_token_auths "
+                    "where validationSession = ? and token = ?",
```

Comment:
Can we use multi line string literals please, and can SQL keywords be in caps?

Reply:
I'm following the surrounding examples, but yeah doesn't hurt to start doing the right then.